docs/guide/advanced/navigation-guards.md

```diff
@@ -28,6 +28,15 @@ And can optionally return any of the following values:
 - `false`: cancel the current navigation. If the browser URL was changed (either manually by the user or via back button), it will be reset to that of the `from` route.
 - A [Route Location](../../api/#routelocationraw): Redirect to a different location by passing a route location as if you were calling [`router.push()`](../../api/#push), which allows you to pass options like `replace: true` or `name: 'home'`. The current navigation is dropped and a new one is created with the same `from`.
 
+```js
+router.beforeEach(async (to, from) => {
+  if (!someConditional(to)) return {
+    name: 'AlternativeRoute',
+    replace: true
+  }
+})
+```
```

Comment:
```suggestion
  ```js
  router.beforeEach(async (to, from) => {
    if (
      // make sure the user is authenticated
      !isAuthenticated &&
      // ❗️ Avoid an infinite redirect
      to.name !== 'Login'
    ) {
      // redirect the user to the login page
      return { name: 'Login' }
    }
  })
  ```
```